### ISSUE TITLE
Analysis Server error / Unexpected exception while performing ResolveDirectiveElementsTask

### ISSUE THREAD

**Issue by MikeMitterer** [dart-lang/sdk#28686]
I copied the line 48-49 and pasted it a line below - now 51-52. This crashed the server (filename: main.dart)

![capturfiles-20170208_040058](https://cloud.githubusercontent.com/assets/116654/22742669/d3914e40-ee17-11e6-8189-f51bde3aa15d.png)

I don't know if this could have an influence but most of the time I have 2 Dart-Applications (browser based) running in parallel. The first app runs e.g. on port 8000 the second one runs on port 7000

[report.txt](https://github.com/dart-lang/sdk/files/760972/report.txt)
Analyzer Feedback from IntelliJ

## Version information

- `IDEA 171.2822.15`
- `1.21.1`
- `IU-171.2822.15, JRE 1.8.0_112-release-b702x64 JetBrains s.r.o, OS Mac OS X(x86_64) v10.11.6 unknown, screens 1920x1080, 1920x1080, 2560x1440`

## Exception

Please append the contents of:
file:///private/var/folders/m4/9rg16xhd2rlbln753n9zk1540000gn/T/report.txt

```
Dart analysis server, SDK version 1.21.1, server version 1.17.0, error: Failed to handle completion domain request: {clientRequestTime: 1486565889909, params: {file: /Volumes/Daten/DevLocal/DevDart/MobiCast.Client/web/main.dart, offset: 1805}, method: completion.getSuggestions, id: 5060}
AnalysisException: failed to resolve declarations
Caused by Unexpected exception while performing ResolveDirectiveElementsTask for source /Volumes/Daten/DevLocal/DevDart/MaterialDesignLite/lib/mdltemplate.dart
#0      AnalysisTask._safelyPerform (package:analyzer/task/model.dart:333)
#1      AnalysisTask.perform (package:analyzer/task/model.dart:220)
#2      AnalysisDriver.performWorkItem (package:analyzer/src/task/driver.dart:284)
#3      AnalysisDriver.performAnalysisTask (package:analyzer/src/task/driver.dart:255)
#4      AnalysisContextImpl.performAnalysisTask.<anonymous closure> (package:analyzer/src/context/context.dart:1186)
#5      _PerformanceTagImpl.makeCurrentWhile (package:analyzer/src/generated/utilities_general.dart:189)
#6      AnalysisContextImpl.performAnalysisTask (package:analyzer/src/context/context.dart:1184)
#7      PerformAnalysisOperation.perform (package:analysis_server/src/operation/operation_analysis.dart:362)
#8      AnalysisServer.performOperation (package:analysis_server/src/analysis_server.dart:972)
#9      Future.Future.<anonymous closure> (dart:async/future.dart:118)
#10     _rootRun (dart:async/zone.dart:1146)
#11     _CustomZone.run (dart:async/zone.dart:1026)
#12     _CustomZone.runGuarded (dart:async/zone.dart:924)
#13     _CustomZone.bindCallback.<anonymous closure> (dart:async/zone.dart:951)
#14     _rootRun (dart:async/zone.dart:1150)
#15     _CustomZone.run (dart:async/zone.dart:1026)
#16     _CustomZone.runGuarded (dart:async/zone.dart:924)
#17     _CustomZone.bindCallback.<anonymous closure> (dart:async/zone.dart:951)
#18     Timer._createTimer.<anonymous closure> (dart:async-patch/timer_patch.dart:16)
#19     _Timer._runTimers (dart:isolate-patch/timer_impl.dart:385)
#20     _Timer._handleMessage (dart:isolate-patch/timer_impl.dart:414)
#21     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:148)

Caused by NoSuchMethodError: The getter 'imports' was called on null.
Receiver: null
Tried calling: imports
#0      Object._noSuchMethod (dart:core-patch/object_patch.dart:44)
#1      Object.noSuchMethod (dart:core-patch/object_patch.dart:47)
#2      DirectiveResolver.visitImportDirective (package:analyzer/src/generated/resolver.dart:2312)
#3      ImportDirectiveImpl.accept (package:analyzer/src/dart/ast/ast.dart:5948)
#4      DirectiveResolver.visitCompilationUnit (package:analyzer/src/generated/resolver.dart:2292)
#5      CompilationUnitImpl.accept (package:analyzer/src/dart/ast/ast.dart:2516)
#6      ResolveDirectiveElementsTask.internalPerform (package:analyzer/src/task/dart.dart:5029)
#7      AnalysisTask._safelyPerform (package:analyzer/task/model.dart:321)
#8      AnalysisTask.perform (package:analyzer/task/model.dart:220)
#9      AnalysisDriver.performWorkItem (package:analyzer/src/task/driver.dart:284)
#10     AnalysisDriver.performAnalysisTask (package:analyzer/src/task/driver.dart:255)
#11     AnalysisContextImpl.performAnalysisTask.<anonymous closure> (package:analyzer/src/context/context.dart:1186)
#12     _PerformanceTagImpl.makeCurrentWhile (package:analyzer/src/generated/utilities_general.dart:189)
#13     AnalysisContextImpl.performAnalysisTask (package:analyzer/src/context/context.dart:1184)

**Comment by MikeMitterer** [comment]
Had this crash again:

Analyzer Feedback from IntelliJ

## Version information

- `IDEA 171.3019.7`
- `1.21.1`
- `IU-171.3019.7, JRE 1.8.0_112-release-b702x64 JetBrains s.r.o, OS Mac OS X(x86_64) v10.11.6 unknown, screens 1920x1080, 1920x1080, 2560x1440`

## Exception

Please append the contents of:
file:///private/var/folders/m4/9rg16xhd2rlbln753n9zk1540000gn/T/report.txt

```
Dart analysis server, SDK version 1.21.1, server version 1.17.0, error: Task failed: ResolveDirectiveElementsTask for source /Volumes/Daten/DevLocal/DevDart/MaterialDesignLite/lib/mdlobservable.dart
Unexpected exception while performing ResolveDirectiveElementsTask for source /Volumes/Daten/DevLocal/DevDart/MaterialDesignLite/lib/mdlobservable.dart
#0      AnalysisTask._safelyPerform (package:analyzer/task/model.dart:333)
#1      AnalysisTask.perform (package:analyzer/task/model.dart:220)
#2      AnalysisDriver.performWorkItem (package:analyzer/src/task/driver.dart:284)
#3      AnalysisDriver.performAnalysisTask (package:analyzer/src/task/driver.dart:255)
#4      AnalysisContextImpl.performAnalysisTask.<anonymous closure> (package:analyzer/src/context/context.dart:1186)
#5      _PerformanceTagImpl.makeCurrentWhile (package:analyzer/src/generated/utilities_general.dart:189)
#6      AnalysisContextImpl.performAnalysisTask (package:analyzer/src/context/context.dart:1184)
#7      PerformAnalysisOperation.perform (package:analysis_server/src/operation/operation_analysis.dart:362)
#8      AnalysisServer.performOperation (package:analysis_server/src/analysis_server.dart:972)
#9      Future.Future.<anonymous closure> (dart:async/future.dart:118)
#10     _rootRun (dart:async/zone.dart:1146)
#11     _CustomZone.run (dart:async/zone.dart:1026)
#12     _CustomZone.runGuarded (dart:async/zone.dart:924)
#13     _CustomZone.bindCallback.<anonymous closure> (dart:async/zone.dart:951)
#14     _rootRun (dart:async/zone.dart:1150)
#15     _CustomZone.run (dart:async/zone.dart:1026)
#16     _CustomZone.runGuarded (dart:async/zone.dart:924)
#17     _CustomZone.bindCallback.<anonymous closure> (dart:async/zone.dart:951)
#18     Timer._createTimer.<anonymous closure> (dart:async-patch/timer_patch.dart:16)
#19     _Timer._runTimers (dart:isolate-patch/timer_impl.dart:385)
#20     _Timer._handleMessage (dart:isolate-patch/timer_impl.dart:414)
#21     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:148)

Caused by NoSuchMethodError: The getter 'imports' was called on null.
Receiver: null
Tried calling: imports
#0      Object._noSuchMethod (dart:core-patch/object_patch.dart:44)
#1      Object.noSuchMethod (dart:core-patch/object_patch.dart:47)
#2      DirectiveResolver.visitImportDirective (package:analyzer/src/generated/resolver.dart:2312)
#3      ImportDirectiveImpl.accept (package:analyzer/src/dart/ast/ast.dart:5948)
#4      DirectiveResolver.visitCompilationUnit (package:analyzer/src/generated/resolver.dart:2292)
#5      CompilationUnitImpl.accept (package:analyzer/src/dart/ast/ast.dart:2516)
#6      ResolveDirectiveElementsTask.internalPerform (package:analyzer/src/task/dart.dart:5029)
#7      AnalysisTask._safelyPerform (package:analyzer/task/model.dart:321)
#8      AnalysisTask.perform (package:analyzer/task/model.dart:220)
#9      AnalysisDriver.performWorkItem (package:analyzer/src/task/driver.dart:284)
#10     AnalysisDriver.performAnalysisTask (package:analyzer/src/task/driver.dart:255)
#11     AnalysisContextImpl.performAnalysisTask.<anonymous closure> (package:analyzer/src/context/context.dart:1186)
#12     _PerformanceTagImpl.makeCurrentWhile (package:analyzer/src/generated/utilities_general.dart:189)
#13     AnalysisContextImpl.performAnalysisTask (package:analyzer/src/context/context.dart:1184)
#14     PerformAnalysisOperation.perform (package:analysis_server/src/operation/operation_analysis.dart:362)
#15     AnalysisServer.performOperation (package:analysis_server/src/analysis_server.dart:972)
[report.txt](https://github.com/dart-lang/sdk/files/763748/report.txt)

**Comment by natebosch** [comment]
Unfortunately we won't be able to reproduce very old analyzer stack traces. Please file a new issue if you are seeing analysis server crashes with a recent SDK.
